core/src/main/java/hudson/model/View.java

```diff
@@ -942,23 +942,28 @@ public synchronized void doSubmitDescription( StaplerRequest req, StaplerRespons
      * Subtypes should override the {@link #submit(StaplerRequest)} method.
      */
     @RequirePOST
-    public final synchronized void doConfigSubmit( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException, FormException {
-        checkPermission(CONFIGURE);
-
-        submit(req);
-
-        description = Util.nullify(req.getParameter("description"));
-        filterExecutors = req.getParameter("filterExecutors") != null;
-        filterQueue = req.getParameter("filterQueue") != null;
-
-        rename(req.getParameter("name"));
-
-        getProperties().rebuild(req, req.getSubmittedForm(), getApplicablePropertyDescriptors());
-        updateTransientActions();  
-
-        save();
-
-        FormApply.success("../" + Util.rawEncode(name)).generateResponse(req,rsp,this);
+    public final void doConfigSubmit( StaplerRequest req, StaplerResponse rsp ) throws IOException, ServletException, FormException {
+        final ItemGroup parent = Jenkins.getInstance();
+        synchronized (parent) {
+            synchronized (this) {
+                checkPermission(CONFIGURE);
```

Comment:
After the code review I'm not sure that checkPermission() under lock is a good idea. Several security plugins are known to be a bit slow